cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/AzureMetadataCollector.java

```diff
@@ -129,9 +130,51 @@ private String getLocalityIndicator(Integer faultDomainCount, CloudContext cloud
     }
 
     @Override
-    public List<CloudLoadBalancerMetadata> collectLoadBalancer(AuthenticatedContext ac, List<LoadBalancerType> loadBalancerTypes) {
-        // no-op
-        return Collections.emptyList();
-    }
+    public List<CloudLoadBalancerMetadata> collectLoadBalancer(AuthenticatedContext ac, List<LoadBalancerType> loadBalancerTypes,
+            List<CloudResource> resources) {
+        LOGGER.debug("Collecting Azure load balancer metadata, for cluster {}", ac.getCloudContext().getName());
+
+        List<CloudLoadBalancerMetadata> cloudLoadBalancerMetadata = new ArrayList<>();
+        String resourceGroup = azureUtils.getTemplateResource(resources).getName();
+        AzureClient azureClient = ac.getParameter(AzureClient.class);
+
+        for (LoadBalancerType type : loadBalancerTypes) {
+            String loadBalancerName = AzureLoadBalancer.getLoadBalancerName(type);
+            LOGGER.debug("Attempting to collect metadata for load balancer {}, type {}", loadBalancerName, type);
+            try {
+                String ip = null;
+                if (LoadBalancerType.PUBLIC.equals(type)) {
+                    List<String> publicIps = azureClient.getLoadBalancerIps(resourceGroup, loadBalancerName);
+                    if (publicIps.isEmpty()) {
+                        LOGGER.warn("Unable to find public ip address for load balancer {}", loadBalancerName);
+                    } else {
+                        ip = publicIps.get(0);
+                    }
+                } else if (LoadBalancerType.PRIVATE.equals(type)) {
+                    List<String> privateIps = azureClient.getLoadBalancerPrivateIps(resourceGroup, loadBalancerName);
+                    if (privateIps.isEmpty()) {
+                        LOGGER.warn("Unable to find private ip address for load balancer {}", loadBalancerName);
+                    } else {
+                        ip = privateIps.get(0);
+                    }
+                }
 
+                if (StringUtils.isNotEmpty(ip)) {
+                    CloudLoadBalancerMetadata loadBalancerMetadata = new CloudLoadBalancerMetadata.Builder()
+                        .withType(type)
+                        .withIp(ip)
+                        .withName(loadBalancerName)
+                        .build();
+                    cloudLoadBalancerMetadata.add(loadBalancerMetadata);
+                    LOGGER.debug("Saved metadata for load balancer: {}", loadBalancerMetadata);
+                } else {
+                    LOGGER.warn("Unable to find metadata for load balancer {}.", loadBalancerName);
+                }
+            } catch (RuntimeException e) {
+                LOGGER.debug("Unable to find metadata for load balancer " + loadBalancerName, e);
```

Comment:
The metadata is used later on to register the LB's FQDN. It's not used for anything else. DNS registration failures do not result in an exception being thrown. Instead we just log an error message and continue the stack creation. Since this metadata is only used for DNS registration, the design decision was made that we would follow that pattern, and would simply log a message and continue stack creation.

Reply:
IIRC, when the `cloudLoadBalancerMetadata` isn't filled out correctly we fall back to using the plain IP address for the LB. @hreeve-cloudera is that right?

Which is to say, we can continue provisioning even if the Metadata collection fails.

I'm not sure how the `CloudConnectorException` fits into the broader exception handling -- should we rethrow that so that we can return appropriate warning back to the user?

Reply:
Thanks, then could you pls use LOGGER.warn uniformly in both cases?

Reply:
I've changed it to a warning.